internal/cli/atlas/accesslogs/list.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslogs
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/cli/require"
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/store"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/cobra"
+	atlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	success      = "success"
+	fail         = "fail"
+	listTemplate = `HOSTNAME	CLUSTER NAME	AUTH RESULT	LOG LINE {{range .AccessLogs}}
+{{if .Hostname}}{{.Hostname}} {{else}}N/A{{end}}{{.Hostname}}	{{if .ClusterName}}{{.ClusterName}} {{else}}N/A{{end}}	{{.AuthResult}}	{{.LogLine}}{{end}}
+`
+	missingClusterNameHostnameErrorMessage = "one between --%s and --%s must be set"
+	invalidValueAuthResultErrorMessage     = `--%s must be set to "%s" or "%s"`
+)
+
+type ListOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	hostname    string
+	clusterName string
+	start       string
+	end         string
+	nLogs       int
+	ipAddresses string
+	authResult  string
+	store       store.AccessLogsLister
+}
+
+func (opts *ListOpts) initStore() error {
+	var err error
+	opts.store, err = store.New(store.AuthenticatedPreset(config.Default()))
+	return err
+}
+
+func (opts *ListOpts) Run() error {
+	if opts.clusterName != "" {
+		r, err := opts.store.AccessLogsByClusterName(opts.ConfigProjectID(), opts.clusterName, opts.newAccessLogOptions())
+		if err != nil {
+			return err
+		}
+
+		return opts.Print(r)
+	}
+
+	r, err := opts.store.AccessLogsByHostname(opts.ConfigProjectID(), opts.hostname, opts.newAccessLogOptions())
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func (opts *ListOpts) newAccessLogOptions() *atlas.AccessLogOptions {
+	authResult := true
+	if strings.EqualFold(opts.authResult, fail) {
+		authResult = false
+	}
+
+	return &atlas.AccessLogOptions{
+		Start:      opts.start,
+		End:        opts.end,
+		NLogs:      opts.nLogs,
+		IPAddress:  opts.ipAddresses,
+		AuthResult: &authResult,
+	}
+}
+
+func (opts *ListOpts) ValidateInput() error {
+	if err := opts.ValidateProjectID(); err != nil {
+		return err
+	}
+
+	if opts.clusterName == "" && opts.hostname == "" {
+		return fmt.Errorf(missingClusterNameHostnameErrorMessage, "clusterName", "hostname")
```

Comment:
the important part here is to share the name with the command definition 
```suggestion
		return fmt.Errorf(missingClusterNameHostnameErrorMessage, flag.ClusterName, flag.Hostname)
```

Reply:
ah right! xD

Reply:
updated